.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,233 @@
+          aws cloudwatch put-metric-alarm "${COMMON_API_PARAMETERS[@]}" \
+            --alarm-name 'OTel Python Soak Tests - Virtual Memory Usage Spike' \
+            --alarm-description 'Triggers when the Virtual Memory Usage spikes above the allowed threshold DURING the Soak Test.' \
+            --threshold ${{ env.TOTAL_MEMORY_THRESHOLD }} \
+            --cli-input-json "{ \"Metrics\": $(cat ./.github/aws-cloudwatch-api-values/get-metric-data/prepared-total-memory-metric-data-query.json) }"
+      - name: Run All Docker Containers - Sample App + OTel Collector + Load Generator + Alarm Poller
+        id: check-failure-during-soak-tests
+        continue-on-error: true
+        working-directory: .github/docker-soak-tests
+        env:
+          APP_IMAGE: ${{ steps.docker-image-metadata.outputs.tags }}
+          INSTANCE_ID: ${{ github.run_id }}-${{ github.run_number }}
+          LISTEN_ADDRESS: 0.0.0.0:8080
+          LOG_GROUP_NAME: ${{ github.repository }}/soak-tests
+          LOGS_NAMESPACE: ${{ env.LOGS_NAMESPACE }}
+          LOG_STREAM_NAME: ${{ env.IMAGE_SUFFIX }}
+          PROCESS_COMMAND_LINE: ${{ env.PROCESS_COMMAND_LINE }}
+          PROCESS_EXECUTABLE_NAME: ${{ env.PROCESS_EXECUTABLE_NAME }}
+          HOSTMETRICS_COLLECTION_INTERVAL_IN_SECONDS: ${{ env.HOSTMETRICS_COLLECTION_INTERVAL_IN_SECONDS }}
+          SOAK_TEST_DURATION_IN_MINUTES: ${{ env.SOAK_TEST_DURATION_IN_MINUTES }}
+        run: >-
+          docker-compose up;
+          exit $(docker inspect $(docker ps --quiet --all --filter "name=docker-soak-tests_alarms-poller") --format="{{.State.ExitCode}}");
+      - name: Get a snapshot of metrics
+        run: >-
+          mkdir -p soak-tests/snapshots/${{ github.sha }};
+          for metric_kind in cpu-load total-memory;
+          do
+            aws cloudwatch get-metric-widget-image --metric-widget "$(
+              cat ./.github/aws-cloudwatch-api-values/get-metric-widget-statistics/$metric_kind-metric-widget.json |
+              sed -e "s/<SOAK_TEST_DURATION_IN_MINUTES>/${{ env.SOAK_TEST_DURATION_IN_MINUTES }}/g" \
+                  -e "s/<METRIC_PERIOD>/${{ env.HOSTMETRICS_COLLECTION_INTERVAL_IN_SECONDS }}/g" \
+                  -e "s/<CPU_LOAD_THRESHOLD>/${{ env.CPU_LOAD_THRESHOLD }}/g" \
+                  -e "s/<TOTAL_MEMORY_THRESHOLD>/${{ env.TOTAL_MEMORY_THRESHOLD }}/g" \
+                  -e "s/<NUM_OF_CPUS>/$NUM_OF_CPUS/g" \
+                  -e "s/<LOGS_NAMESPACE>/$(echo ${{ env.LOGS_NAMESPACE }} | sed 's/\//\\\//g')/g" \
+                  -e "s/<PROCESS_COMMAND_LINE_DIMENSION_VALUE>/$(echo ${{ env.PROCESS_COMMAND_LINE }} | sed 's/\//\\\//g')/g"
+            )" |
+            jq -r '.MetricWidgetImage' |
+            base64 -d > soak-tests/snapshots/${{ github.sha }}/${{ matrix.app-platform }}-${{ matrix.instrumentation-type }}-$metric_kind-results-$GITHUB_RUN_ID.png;
+          done
+      - name: Commit snapshots to repository
+        uses: EndBug/add-and-commit@v7
```

Comment:
Don't need an action, just use `git` commands directly

Reply:
Okay will try using `git` commands! I liked how clean this reads compared to using `run:` and a bunch of commands, but I haven't tried it yet so I should.

Reply:
I added it! I'm curious what user the commits will be under? Will run the tests in my demo repo to verify in the meantime.